dcrutil/address.go

```diff
@@ -89,6 +75,34 @@ func encodePKAddress(serializedPK []byte, netID [2]byte, algo dcrec.SignatureTyp
 	return base58.CheckEncode(pubKeyBytes, netID)
 }
 
+// AddressParams defines an interface that is used to provide the parameters
+// required when encoding and decoding addresses.  These values are typically
+// well-defined and unique per network.
+type AddressParams interface {
+	// AddrIDPubKeyV0 returns the magic prefix bytes for version 0 pay-to-pubkey
+	// addresses.
+	AddrIDPubKeyV0() [2]byte
+
+	// AddrIDPubKeyHashECDSAV0 returns the magic prefix bytes for version 0
+	// pay-to-pubkey-hash addresses where the underlying pubkey is secp256k1 and
+	// the signature algorithm is ECDSA.
+	AddrIDPubKeyHashECDSAV0() [2]byte
```

Comment:
This reads a bit strangely due to the capitalized ECDSA and V0 concatenated together.  Unfortunately i don't see a great way around this except by moving the V0.  What do you think about moving it to the start of the method name?

Reply:
I wouldn't mind, but unfortunately it can't easily be changed now because it would make `chaincfg.Params` added in #1766 no longer satisfy the interface.

Reply:
Makes sense and I'm partly to blame here since I approved that.